packages/babel-parser/src/parser/statement.js

```diff
@@ -2144,10 +2145,16 @@ export default class StatementParser extends ExpressionParser {
       }
 
       const node = this.startNode();
-      node.local = this.parseModuleExportName();
-      node.exported = this.eatContextual("as")
-        ? this.parseModuleExportName()
-        : node.local.__clone();
+      const isString = this.match(tt.string);
```

Comment:
This can be inlined in the `else if` condition, so we avoid calling it when there is `"as"`.

Reply:
`parseModuleExportName` will eat the string, so we have to call it first.